src/sentry/static/sentry/app/components/charts/components/tooltip.jsx

```diff
@@ -1,9 +1,70 @@
+import moment from 'moment';
 import 'echarts/lib/component/tooltip';
 
-export default function Tooltip(props = {}) {
+const DEFAULT_TRUNCATE_LENGTH = 80;
+
+// Truncates labels for tooltip
+function truncateLabel(seriesName, truncate) {
+  if (!truncate) {
+    return seriesName;
+  }
+
+  let result = seriesName;
+  let truncateLength = typeof truncate === 'number' ? truncate : DEFAULT_TRUNCATE_LENGTH;
+  0;
+
+  if (seriesName.length > truncateLength) {
+    result = seriesName.substring(0, truncateLength) + '…';
+  }
+  return result;
+}
+
+function formatAxisLabel(value, isTimestamp) {
+  if (!isTimestamp) {
+    return value;
+  }
+
+  return moment(value).format('MMM D, YYYY');
+}
+
+function getFormatter({filterEmpty, filterNull, filter, isGroupedByDate, truncate}) {
```

Comment:
I'd vote to drop `filterEmpty` and `filterNull` since you can easily pass a filter function that does the same. I think there's value in keeping this API simple.

Reply:
https://github.com/getsentry/sentry/pull/9652/commits/34e0c529eeb04f0c3cde93e8d0fe11c91799117d